### PR TITLE
R: fix build with curl >= 8.0.0

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix Build break with bump of curl to 8.0.1

###### Change Log  

- Introduce patch to resolve build break with curl >= 8.0.0

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://github.com/microsoft/CBL-Mariner/pull/5172

###### Links to CVEs  
- NA

###### Test Methodology

- Local Build

- Full Build [PR-5194](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=336378&view=results)

NOTE: The bump to curl 8.0.0 is not a real breaking change from 7.x as per [this](https://daniel.haxx.se/blog/2023/03/20/curl-8-0-0-is-here/)

### CODE REVIEW

File: SPECS/R/R.spec [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
 Version:        4.1.0
-Release:        2%{?dist}
+Release:        3%{?dist}
 License:        GPLv2
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Daemons
 URL:            https://www.r-project.org
 Source0:        https://cran.r-project.org/src/base/R-4/R-%{version}.tar.gz
+Patch0:         0001-configure-fix-compilation-with-curl-8.0.0.patch
```
Comment (PawelWMS) [inline SPECS/R/R.spec:12]: Could you only add a comment explaining that this is our best hunch fix and the patch will need to be removed once `R` "officially" supports cURL 8+?

**Review by PawelWMS — APPROVED** [review]
Discussed offline - approving on the assumption ptests succeed.

**Review by anphel31 — APPROVED** [review]

**Comment by mfrw** [comment]
> Discussed offline - approving on the assumption ptests succeed.

Ptests passed locally :)

File: SPECS/R/R.spec [hunk 2]
```diff
@@ -2,13 +2,17 @@
...
-Release:        2%{?dist}
+Release:        3%{?dist}
 License:        GPLv2
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Daemons
 URL:            https://www.r-project.org
 Source0:        https://cran.r-project.org/src/base/R-4/R-%{version}.tar.gz
+# This change is a best hunch fix as the limitation on curl version 7 was set
+# on 2018. Given curl 8.0.0 is not an actual breaking change, this patch should be fine.
```
Comment (mfrw) [inline SPECS/R/R.spec:13]:
```suggestion
# in 2018. Given curl 8.0.0 is not an actual breaking change, this patch should be fine.
```

**Pull request merged**